lms/resources/lti_launch.py

```diff
@@ -78,9 +78,20 @@ def h_group(self):
 
         return course
 
+    @property
+    def _is_speed_grader(self):
+        return bool(self._request.GET.get("learner_canvas_user_id"))
+
     @property
     def resource_link_id(self):
-        return self._request.params.get("resource_link_id")
+        # If we are speed grading take the value from the query params that we injected
+        # ourselves instead of the wrong value provided by canvas in the body.
+        if self._is_speed_grader and (
+            resource_link_id := self._request.GET.get("resource_link_id")
```

Comment:
Just to make sure I checked the WebOb docs and it looks like `request.GET` _is_ an officially supported way to get only the query params:

https://docs.pylonsproject.org/projects/webob/en/stable/api/request.html#webob.request.BaseRequest.GET  
https://docs.pylonsproject.org/projects/webob/en/stable/reference.html#query-post-variables